src/commands/serverActions.ts

```diff
@@ -56,6 +62,44 @@ export async function serverActions(): Promise<void> {
         await checkConnection(true, undefined, true);
         break;
       }
+      case "switchNamespace": {
+        const allNamespaces: string[] | undefined = await api
+          .serverInfo()
+          .then((data) => data.result.content.namespaces)
+          .catch((reason) => {
+            // Notify user about serverInfo failure
+            vscode.window.showErrorMessage(reason.errorText || `Failed to fetch namespace list`, "Dismiss");
+            return undefined;
+          });
+
+        if (!allNamespaces) {
+          return;
+        }
+        // Handle serverInfo having returned no namespaces
+        if (!allNamespaces.length) {
+          vscode.window.showErrorMessage(`No namespace list returned`, "Dismiss");
+          return;
+        }
+
+        const namespace = await vscode.window.showQuickPick(allNamespaces, {
+          placeHolder: `Namespace on server `,
```

Comment:
I suggest "Choose the namespace to switch to"

Reply:
Done